source/neuropod/bindings/java/BUILD

```diff
@@ -0,0 +1,170 @@
+      "@libtorch_repo//:libtorch_libs",
+    ] ,
+    visibility = [
+        "//neuropod:__subpackages__",
+    ],
+)
+
+filegroup(
+    name = "neuropod_jni",
+    srcs = select({
+        "@bazel_tools//src/conditions:darwin": [":libneuropod_jni.dylib"],
+        "//conditions:default": [":libneuropod_jni.so"],
+        }),
+    visibility = [
+        "//neuropod:__subpackages__",
+    ],
+)
+
+filegroup(
+    name = "java_source",
+    srcs = glob(["src/main/java/com/uber/neuropod/*.java"]),
+    visibility = [
+        "//neuropod:__subpackages__",
+    ],
+)
+
+java_binary(
+    name = "neuropod_java_jar",
+    srcs = [":java_source"],
+    # TODO: Find a way to make these native libraries locate at /com/uber/neuropod/native/{OS}/{PLATFORM}
+    # in the jar file, instead of locating at the class root path.
+    classpath_resources = [
+        "//neuropod:libneuropod.so",
+        "//neuropod/multiprocess:neuropod_multiprocess_worker",
+        ":neuropod_jni",
+        ],
+    javacopts = JAVACOPTS,
+    visibility = ["//visibility:public"],
+)
+
+java_library(
+    name = "neuropod_java",
+    srcs = [":java_source"],
+    data = [
+        ":backends",
+        "//neuropod:libneuropod.so",
+        "//neuropod/multiprocess:neuropod_multiprocess_worker",
+        ":neuropod_jni",
+        ],
+    javacopts = JAVACOPTS,
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "copy_jni_hdr_lib",
+    hdrs = [
+        ":copy_link_jni_header",
+        ":copy_link_jni_md_header",
+    ],
+    includes = ["."],
+)
+
+cc_library(
+    name = "jni_core",
+    hdrs = glob(["src/main/native/*.h"]),
+    includes = select({
+        "@bazel_tools//src/conditions:darwin": [ "external/local_jdk/include/darwin" ],
+        "//conditions:default": [ "external/local_jdk/include/linux" ],
+    }),
+    visibility = [
+        "//neuropod:__subpackages__",
+    ],
+    deps = [
+        ":copy_jni_hdr_lib",
+        "//neuropod:neuropod_hdrs",
+    ],
+)
+
+# Use for linux only. A naive way to create different filename
+# for different OS.
+cc_binary(
+    name = "libneuropod_jni.so",
+    srcs = glob(["src/main/native/*.cc"]) + [
+        "//neuropod:libneuropod.so",
+    ],
+
+    linkopts = select({
+        "//conditions:default": [
+            "-Wl,-rpath,$$ORIGIN",
+            ],
+    }),
+    linkshared = True,
+    visibility = [
+        "//visibility:public",
+    ],
+    deps = [
+        ":jni_core",
+    ],
+)
+
+# Use for darwin only. A naive way to create different filename
+# for different OS.
+cc_binary(
+    name = "libneuropod_jni.dylib",
+    srcs = glob(["src/main/native/*.cc"]) + [
+        "//neuropod:libneuropod.so",
+    ],
+
+    linkopts = select({
+        "@bazel_tools//src/conditions:darwin": ["-Wl,-rpath,@loader_path"],
+    }),
+    linkshared = True,
+    visibility = [
+        "//visibility:public",
+    ],
+    deps = [
+        ":jni_core",
+    ],
+)
```

Comment:
This approach isn't going to work on linux (because it's going to attempt to build something named `libneuropod_jni.dylib` and fail (see the output of the failing CI for this PR).

Another solution that might work is to just use a genrule for the dylib (since we just care about changing the name). Maybe something like:

```
genrule(
    name = "copy_to_dylib",
    srcs = [":libneuropod_jni.so"],
    outs = ["libneuropod_jni.dylib"],
    cmd = "cp -f $< $@",
)
```

This isn't ideal, but we can improve it after we get a basic first version running. Ideally, we'd be able to provide the full name to `System.loadLibrary` (e.g. `libneuropod_jni.so`) and not have to do this during the build